src/main/java/tigase/server/Packet.java

```diff
@@ -598,6 +601,44 @@ public void setPriority(Priority priority) {
 		return processorsIds;
 	}
 
+	/**
+	 * Method returns JID from-address that was authenticated and bound to user session.
+	 *
+	 * This is a helper method that facilitates and improve maintaining correct stanza from when
+	 * communicating between ClientConnectionManager and SessionManager.
+	 *
+	 * This is a temporary solution! In version 9.0, after reviewing the APIs and correcting clustering strategy
+	 * packets incomming in (Client) Connection Managers should already have correct 'from' attribute in stanzas
+	 * set in {@code tigase.server.xmppclient.ClientConnectionManager#processSocketData(tigase.xmpp.XMPPIOService)}.
+	 *
+	 * @return {@code Optional} that may contain user JID if the session was already authorised.
+	 */
+	@Deprecated
+	@TigaseDeprecated(since = "8.2.0", removeIn = "9.0.0")
+	public Optional<JID> getServerAuthorisedStanzaFrom() {
+		return Optional.ofNullable(serverAuthorisedStanzaFrom);
+	}
+
+	/**
+	 * Method used to set JID from-address that was authenticated and bound to user session.
+	 *
+	 * This is a helper method that facilitates and improve maintaining correct stanza from when
+	 * communicating between ClientConnectionManager and SessionManager.
+	 *
+	 * This is a temporary solution! In version 9.0, after reviewing the APIs and correcting clustering strategy
+	 * packets incomming in (Client) Connection Managers should already have correct 'from' attribute in stanzas
+	 * set in {@code tigase.server.xmppclient.ClientConnectionManager#processSocketData(tigase.xmpp.XMPPIOService)}.
+	 *
+	 * @param serverAuthorisedStanzaFrom
+	 */
+	@Deprecated
+	@TigaseDeprecated(since = "8.2.0", removeIn = "9.0.0")
+	public void setServerAuthorisedStanzaFrom(JID serverAuthorisedStanzaFrom) {
```

Comment:
Will this method be called by StreamManagement if resumption failed and it's sending errors? I'm not sure that it will be needed but it would be good to check.

Reply:
But... those packets would be "to" user connection and if they were failed to be delivered, even though they would come from CM, it wouldn't come "from" that user? Besides, those would be handled by `tigase.server.ConnectionManager#processUndeliveredPacket`? And I'm not sure we should be setting `serverAuthorisedStanzaFrom` there.

Reply:
Yes, you are right. This should be handled properly without setting this value.